packages/cli/test/integration/evaluation/test-definitions.api.test.ts

```diff
@@ -0,0 +1,296 @@
+			name: 'updated-test',
+			evaluationWorkflowId: evaluationWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.name).toBe('updated-test');
+		expect(resp.body.data.evaluationWorkflowId).toBe(evaluationWorkflow.id);
+	});
+
+	test('should return error if user has no access to the evaluation workflow', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			name: 'updated-test',
+			evaluationWorkflowId: otherWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(400);
+		expect(resp.body.message).toBe('User does not have access to the evaluation workflow');
+	});
+
+	test('should ignore the provided workflowId', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			name: 'updated-test',
+			workflowId: otherWorkflow.id,
+		});
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.name).toBe('updated-test');
+		expect(resp.body.data.workflowId).toBe(workflowUnderTest.id);
+	});
+
+	test('should update annotationTagId', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			annotationTagId: annotationTag.id,
+		});
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.annotationTag.id).toBe(annotationTag.id);
+	});
+
+	test('should return error if annotationTagId is invalid', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.patch(`/evaluation/test-definitions/${newTest.id}`).send({
+			annotationTagId: 123,
+		});
+
+		expect(resp.statusCode).toBe(400);
+		expect(resp.body.message).toBe('Annotation tag not found');
+	});
+});
+
+describe('DELETE /evaluation/test-definitions/:id', () => {
+	test('should delete test definition', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.delete(`/evaluation/test-definitions/${newTest.id}`);
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.success).toBe(true);
+	});
```

Comment:
Maybe a test case so that user can't delete test definition they don't have access to?